processor/deltatocumulativeprocessor/internal/delta/delta.go

```diff
@@ -96,6 +104,16 @@ func AccumulateInto[P data.Point[P]](state P, dp P) error {
 		return ErrOutOfOrder{Last: state.Timestamp(), Sample: dp.Timestamp()}
 	}
 
-	state.Add(dp)
+	switch dp := any(dp).(type) {
+	case pmetric.NumberDataPoint:
+		state := any(state).(pmetric.NumberDataPoint)
+		data.Number{NumberDataPoint: state}.Add(data.Number{NumberDataPoint: dp})
+	case pmetric.HistogramDataPoint:
+		state := any(state).(pmetric.HistogramDataPoint)
+		data.Histogram{HistogramDataPoint: state}.Add(data.Histogram{HistogramDataPoint: dp})
+	case pmetric.ExponentialHistogramDataPoint:
+		state := any(state).(pmetric.ExponentialHistogramDataPoint)
+		data.ExpHistogram{DataPoint: state}.Add(data.ExpHistogram{DataPoint: dp})
+	}
```

Comment:
This refactor effectively eliminates the need for the data package, as we no longer rely on type characteristics.

I'll refactor datapoint addition in a future PR, making this part more clear, maybe like this:

```go
var add data.Aggregator = new(data.Add)

switch into := any(dp).(type) {
case pmetric.NumberDataPoint:
	add.Numbers(into, dp)
case pmetric.HistogramDataPoint:
	add.Histograms(into, dp)
}
```